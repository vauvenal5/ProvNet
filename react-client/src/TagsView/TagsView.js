import React from 'react';

import { Segment, Label, Header } from 'semantic-ui-react';

export const TagsView = ({tags}) => {

    let labels = [];

    for(let key in tags) {
        let tag = tags[key];
        labels.push(
            <Label color={tag.getColor()} key={tag.id}>
                {tag.getTitle()}
            </Label>
        );
    }

    return (
        <Segment>
            <Label.Group circular>
                <Header as="h2" dividing>Link Types:</Header>
                {labels}
            </Label.Group>
        </Segment>
    );
}

export default TagsView;