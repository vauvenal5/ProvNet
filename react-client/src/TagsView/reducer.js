const tagsViewReducer = (state={selectedContract: {tags: [{title: "trusted"},{title: "known"},{title: "blablu"}]}}, action) => {
    return state;
}

export default tagsViewReducer;