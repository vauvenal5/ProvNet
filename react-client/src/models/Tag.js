import _ from 'lodash';

const colors = [
    'red',
    'orange',
    'yellow',
    'olive',
    'green',
    'teal',
    'blue',
    'violet',
    'purple',
    'pink',
    'brown',
    'grey',
    'black',
];

export default class Tag {
    constructor(id, title) {
        this.id = id;
        this.title = title;
    }

    getColor() {
        return colors[(this.id-1)%colors.length];
    }

    getTitle() {
        return _.capitalize(this.title);
    }
}