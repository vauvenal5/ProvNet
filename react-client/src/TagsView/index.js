import Component from "./TagsView";
import reducer from "./reducer";

export default {Component, reducer};