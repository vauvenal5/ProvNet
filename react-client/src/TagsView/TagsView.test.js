import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TagsView from "./TagsView";
import Tag from '../models/Tag';

Enzyme.configure({ adapter: new Adapter() });

const setup = (propOverrides) => {
    const props = Object.assign({}, propOverrides);

    const wrapper = shallow(<TagsView {...props}/>);

    return {
        props,
        wrapper,
        render: wrapper.debug()
    }
}

describe("TagsView", () => {
    it("should render empty view", () => {
        const { render } = setup();
        expect(render).toMatchSnapshot();
    });

    it("should render with tags", () => {
        const props = {
            tags: {
                1: new Tag(1, "trusted"),
                2: new Tag(2, "known")
            }
        }
        const {render} = setup(props);
        expect(render).toMatchSnapshot();
    });
});