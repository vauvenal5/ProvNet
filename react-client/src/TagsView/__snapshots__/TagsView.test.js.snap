// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsView should render empty view 1`] = `
"<Segment>
  <LabelGroup circular={true}>
    <Header as=\\"h2\\" dividing={true}>
      Link Types:
    </Header>
  </LabelGroup>
</Segment>"
`;

exports[`TagsView should render with tags 1`] = `
"<Segment>
  <LabelGroup circular={true}>
    <Header as=\\"h2\\" dividing={true}>
      Link Types:
    </Header>
    <Label color=\\"red\\">
      Trusted
    </Label>
    <Label color=\\"orange\\">
      Known
    </Label>
  </LabelGroup>
</Segment>"
`;
